File: backend/user_details/enums/diet_intensity.py
from enum import Enum


class DietIntensity(str, Enum):
    SLOW = "slow"
    NORMAL = "normal"
    FAST = "fast"

    def get_intensity_factor(self):
        return {
            DietIntensity.SLOW: 0.1,
            DietIntensity.NORMAL: 0.15,
            DietIntensity.FAST: 0.25,
        }[self]

File: frontend/lib/models/user_details/diet_intensity.dart
enum DietIntensity {
  slow(0, 'SLOW'),
  medium(1, 'MEDIUM'),
  fast(2, 'FAST');

  final int value;
  final String nameStr;

  const DietIntensity(this.value, this.nameStr);

  String toJson() => nameStr;

  static DietIntensity fromJson(String value) {
    return DietIntensity.values.firstWhere(
          (e) => e.nameStr == value.toUpperCase(),
      orElse: () => throw ArgumentError('Unknown diet intensity: $value'),
    );
  }

  int toInt() => value;

  static DietIntensity fromInt(int value) {
    return DietIntensity.values.firstWhere(
          (e) => e.value == value,
      orElse: () => throw ArgumentError('Invalid diet intensity value: $value'),
    );
  }
}
